nixos/modules/services/audio/spotifyd.nix

```diff
@@ -0,0 +1,46 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.spotifyd;
+  spotifydConf = pkgs.writeText "spotifyd.conf" cfg.config;
+in
+{
+  options = {
+    services.spotifyd = {
+      enable = mkEnableOption "spotifyd, a Spotify playing daemon";
+
+      dataDir = mkOption {
+        default = "spotifyd";
+        type = types.str;
+        description = "The directory where spotifyd stores its state, relative to /var/lib/.";
+      };
+
+      config = mkOption {
```

Comment:
could this be made an attrset instead, which gets assembled into config below the `[global]` section?

Reply:
Or maybe just generate the `ini` with `lib.generators.toINI`?

Reply:
yeah, that's what I meant :-)